### PR TITLE
Add readerwriterqueue/1.0.3

### PR DESCRIPTION
**Pull request by hnOsmium0001**
Specify library name and version:  **readerwriterqueue/1.0.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/readerwriterqueue/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    homepage = "https://github.com/cameron314/readerwriterqueue"
+    description = "A fast single-producer, single-consumer lock-free queue for C++"
+    topics = ("cpp11", "cpp14", "cpp17", "queue", "lock-free")
+    license = "BSD-2-Clause"
+    no_copy_source = True
+    settings = "os"
+    
+    @property
+    def _source_subfolder(self):
+        return "readerwriterqueue"
```
Comment (prince-chrismc) [inline recipes/readerwriterqueue/all/conanfile.py:18]:
```suggestion
        return "sources_subfolder"
```

File: recipes/readerwriterqueue/all/test_package/example.cpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#include <cassert>
+#include <moodycamel/readerwriterqueue.h>
+#include <thread>
+
+int main(int argc, char **argv) {
+  moodycamel::ReaderWriterQueue<int> q;
```
Comment (intelligide) [inline recipes/readerwriterqueue/all/test_package/example.cpp:6]:
```suggestion
  moodycamel::ReaderWriterQueue<int> q;
  q.enqueue(42); 
```

File: recipes/readerwriterqueue/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        return "sources_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("readerwriterqueue-{}".format(self.version),
+                  self._source_subfolder)
+
+    def package(self):
+        self.copy("atomicops.h", src=self._source_subfolder, dst="include/moodycamel")
+        self.copy("readerwriterqueue.h", src=self._source_subfolder, dst="include/moodycamel")
```
Comment (prince-chrismc) [inline recipes/readerwriterqueue/all/conanfile.py:27]: is the `moodycamel` part of the structure of the project?

Comment (uilianries) [inline recipes/readerwriterqueue/all/conanfile.py:27]:
```suggestion
        self.copy("atomicops.h", src=self._source_subfolder, dst=os.path.join("include", "moodycamel"))
        self.copy("readerwriterqueue.h", src=self._source_subfolder, dst=os.path.join("include", "moodycamel"))
```

**Review by uilianries — CHANGES_REQUESTED** [review]

File: recipes/readerwriterqueue/all/conanfile.py (hunk 3 above)
Comment (hnOsmium0001) [reply recipes/readerwriterqueue/all/conanfile.py:27]: I was simply following a similar package (from the same author)'s path:
https://github.com/conan-io/conan-center-index/blob/master/recipes/concurrentqueue/all/conanfile.py#L25

If we correct it to `include/xxx.h`, I think we should change that one as well in a new version or something.

Comment (prince-chrismc) [reply recipes/readerwriterqueue/all/conanfile.py:27]: From the README  https://github.com/cameron314/readerwriterqueue#cmake-installation

> Then, you can include it from your source code: `#include <readerwriterqueue/readerwriterqueue.h>`

You just need to fix the name

the other mine seems to be a mistake https://github.com/cameron314/concurrentqueue#basic-use

File: recipes/readerwriterqueue/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        self.run(os.path.join("bin", "test_package"), run_environment=True)
```
Comment (madebr) [inline recipes/readerwriterqueue/all/test_package/conanfile.py:14]: Even though the package is header only, the host system might still be non compatible with the build system.
```suggestion
        if not tools.cross_building(self.settings):
            self.run(os.path.join("bin", "test_package"), run_environment=True)
```

**Review by prince-chrismc — APPROVED** [review]
👍

File: recipes/readerwriterqueue/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+from conans import ConanFile, CMake, tools
+from fnmatch import fnmatch
```
Comment (uilianries) [inline recipes/readerwriterqueue/all/conanfile.py:2]:
```suggestion
```

Not used

File: recipes/readerwriterqueue/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/readerwriterqueue/all/conanfile.py:1]:
```suggestion
from conans import ConanFile, tools
```

Not used

**Review by uilianries — CHANGES_REQUESTED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Pull request merged**